packages/firestore/test/unit/local/simple_db.test.ts

```diff
@@ -128,11 +133,13 @@ describe('SimpleDb', () => {
   });
 
   it('lets you explicitly abort transactions', async () => {
-    await runTransaction((store, txn) => {
-      return store.put(dummyUser).next(() => {
-        txn.abort(); // JUST KIDDING!
-      });
-    });
+    expect(
+      runTransaction((store, txn) => {
+        return store.put(dummyUser).next(() => {
+          txn.abort();
+        });
+      })
+    ).to.eventually.be.rejectedWith('The IndexedDb transaction was aborted.');
```

Comment:
FWIW, I don't think aborting a transaction should generally be an error case, so I'd maybe add a todo...

// TODO(mrschmidt): If in the future we end up adding code that intentionally aborts transactions we may need to turn this into a non-error case.

Reply:
Done.

Reply:
Per my other comment, I think we can make this work again without generating an error?

Reply:
Yeah, with the conditional check for `transaction.error`, this once again becomes possible.